build.sbt

```diff
@@ -376,6 +376,26 @@ lazy val `sdk-exporter-proto` =
       Compile / PB.targets ++= Seq(
         scalapb.gen(grpc = false) -> (Compile / sourceManaged).value / "scalapb"
       ),
+      Compile / PB.generate := {
+        val files = (Compile / PB.generate).value
+
+        files.filter(_.isFile).foreach { file =>
+          val content = IO.read(file)
+
+          val next = content
+            .replaceAll("io\\.opentelemetry\\.proto", "org.typelevel.otel4s.sdk.exporter.proto")
+            .replaceAll("(?m)^final case class", "private[exporter] final case class")
+            .replaceAll("(?m)^object", "private[exporter] object")
+            .replaceAll("(?m)^sealed abstract", "private[exporter] sealed abstract")
+            .replaceAll("(?m)^sealed trait", "private[exporter] sealed trait")
+            .replaceAll("(?m)^trait", "private[exporter] trait")
+            .replaceAll("(?m)^abstract", "private[exporter] abstract")
+
+          IO.write(file, next)
+        }
+
+        files
+      },
```

Comment:
Is there any better way to do this? Since the models are generated from third-party sources, I cannot find a simple way to customize the generation process.

The file-scope `package_name` option doesn't seem to work https://scalapb.github.io/docs/customizations#file-level-options. And the package-scope https://scalapb.github.io/docs/customizations/#package-scoped-options doesn't have this option.

Reply:
This entry from the FAQ https://scalapb.github.io/docs/faq#how-do-i-mark-a-generated-case-class-private is probably not helpful because you don't have control over the proto files 😢